features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,360 @@
+## Stories
+
+Is it reasonable to break these down
+one per endpoint?
+
+## Data
+
+```js
+{
+  // The user this device belongs to
+  "uid": "4c352927cd4f4a4aa03d7d1893d950b8",
+
+  // The active session for this device
+  "sessionToken": "27cd4f4a4aa03d7d186a2ec81cbf19d5c8a604713362df9ee15c4f4a4aa03d7d",
+
+  // A server assigned identifier for this device
+  "id":"01ab",
+
+  // A human identifier for this devices. Able to be updated by the user
+  "name": "My Phone",
+
+  // General type of device. Useful for device list icons and demographic metrics
+  "type": "mobile",
+
+  // Timestamp of when the device was first created, à la Date.now()
+  "createdAt": 1442785364807,
+
+  // The push notification endpoint for this device/user
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+```
+
+## API
+
+### POST /v1/account/login
+
+Authenticated with session token or OAuth.
+
+#### Requests
+
+##### New device
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device \
+-d '{
+  "name": "My Phone",
+  "type": "mobile",
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+}'
+```
+
+##### Registered device
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device \
+-d '{
+  "id": "3d7d",
+  "name": "My Old Phone"
+}'
+```
+
+#### Response
+
+```
+{
+  "id": "3d7d",
+  "name": "My Phone",
+  "type": "mobile",
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+}
+```
+
+### POST /v1/account/create
+
+Authenticated with session token or OAuth.
+
+#### Requests
+
+##### New device
+
+```
+curl -v \
+-X POST \
+-H "Content-Type: application/json" \
+"https://api-accounts.dev.lcip.org/v1/account/create?keys=true" \
+-d '{
+  "email": "me@example.com",
+  "authPW": "996bc6b1aa63cd69856a2ec81cbf19d5c8a604713362df9ee15c2bf07128efab",
+  "device": {
+    "name": "My Phone",
+    "type": "mobile",
+    "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+  }
+}'
+```
+
+##### Registered device
+
+If a device id exists for this device,
+the request MAY include it
+but it MUST also include the other info as well,
+since device ids are not globally unique.
```

Comment:
As per [Ryan's comment](https://github.com/mozilla/fxa/pull/69/files#r43348663), this is not the case any more. Have updated.